Makefile.tools

```diff
@@ -88,7 +84,13 @@ $(KUSTOMIZE) kustomize: $(TOOLS_DIR)
 $(OPERATOR_SDK) operator-sdk: $(TOOLS_DIR)
 	@{ \
 		set -ex ;\
-		[[ -f $(OPERATOR_SDK) ]] && exit 0 ;\
+		if [ -f $(OPERATOR_SDK) ]; then \
+			if [ $(shell $(OPERATOR_SDK) version | awk '{print $$3}' | tr -d ,) = $(OPERATOR_SDK_VERSION) ]; then \
+				echo "sdk up to date"; \
+				exit 0 ;\
+			fi; \
+			echo "sdk needs update" ;\
+		fi; \
```

Comment:
We can simplify nested using `&&` and totally do away with `if` e.g. the snippet below works 


```suggestion
		[[ -x $(OPERATOR_SDK) ]] && \
		[[  "$$( $(OPERATOR_SDK) version | awk '{print $$3}' | tr -d , )" == \"$(OPERATOR_SDK_VERSION)\" ]] && { \
				echo "operator-sdk up to date"; \
				exit 0 ;\
		}; \
```

I think `[[ -x ` is better than `[[ -f`

Reply:
- Not sure we should move to `-x`. Directories can be executable as well. We really only want to check if a file exists so `-e` would probably me most appropriate. `-f` ensures its a regular file, if this is not executable, I think the error message from trying to execute it should suffice?
- `[[ ]]` vs `if`, you got a point for consistency's sake. Generally I'd be more inclined to be more explicit about the logic structure. But this will unlikely change a lot, and if it does I'd probably argue for not using bash anymore. This change was already too tricky to get right really.

Reply:
case in point, this already does not work on macos... _sigh_

Reply:
> Not sure we should move to -x. Directories can be executable as well. We really only want to check if a file exists so -e would probably me most appropriate. -f ensures its a regular file, if this is not executable, I think the error message from trying to execute it should suffice?

`./dir` gets you `./dir is a directory` whereas `./file` tells you `./tmp/bin/foo: Permission denied` and hence ` -x ` but I agree we don't have to fret on this 

> [[ ]] vs if, you got a point for consistency's sake

I don't have strong opinion on this. I was pointing at avoiding nested if by using `&&` 
```
if [[ -f foo ]] && [[ "$(./foo | awk .. )" == "bar" ]]; then
  
fi
```


Which should work on bash 3.2 as well.

Reply:
> case in point, this already does not work on macos... sigh

```
bash: -c: line 3: conditional binary operator expected
bash: -c: line 3: syntax error near `v1.23.0'
bash: -c: line 3: `	[[  = v1.23.0 ]] && { \'
```
That seems to because of missing `"` around  `"$$( ... )"`